src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs

```diff
@@ -49,17 +50,26 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             // of the local.  This is necessary in some cases (for example, when the
             // type of the out-var-decl affects overload resolution or generic instantiation).
 
-            foreach (var diagnostic in diagnostics)
+            var originalRoot = editor.OriginalRoot;
+
+            var originalNodes = new (ILocalSymbol, IdentifierNameSyntax, DeclarationExpressionSyntax, SyntaxNode)[diagnostics.Length];
+
+            for (var i = 0; i < originalNodes.Length; i++)
             {
-                cancellationToken.ThrowIfCancellationRequested();
-                await AddEditsAsync(
-                    document, editor, diagnostic, 
-                    options, cancellationToken).ConfigureAwait(false);
+                originalNodes[i] = await GetDeclarationAsync(document, editor, diagnostics[i], options, cancellationToken).ConfigureAwait(false);
             }
+
+            await editor.ApplyExpressionLevelSemanticEditsAsync(
+                document,
+                originalNodes.AsImmutable(),
+                (_1, _2, _3) => true,
+                (semanticModel, currentRoot, t, currentNode) => ReplaceNode(semanticModel, currentRoot, t, currentNode),
+                (t) => t.Item3, // Use the InvocationOrCreation node to track
```

Comment:
item3 is a pretty terrible name.  it also seems incorrect given that Item3 looks to be a DeclarationExpressionSyntax.  Name tuple properties are a good thing :)

Reply:
Aye. I added another item and messed this up :) I'll switch to named tuple properties.